tests/testthat/helper-restart.R

```diff
@@ -0,0 +1,52 @@
+# Example usage of ptype2 and cast restart. This handler treats any
+# input that inherits from <ordered> as a <factor>. In other words, it
+# allows incompatible <ordered> inputs to benefit from all <factor>
+# coercion methods.
+with_ordered_restart <- function(expr) {
+  withCallingHandlers(
+    expr,
+    vctrs_error_incompatible_type = function(cnd) {
+      x <- cnd[["x"]]
+      y <- cnd[["y"]]
+
+      restart <- FALSE
+
+      if (is.ordered(x)) {
+        restart <- TRUE
+        x <- factor(as.character(x), levels = levels(x))
```

Comment:
Annoying that `as.factor()` doesn't strip orderedness, which in theory is really all you need to do here

Reply:
yeah I was surprised

Reply:
Even more horrifying is that it isn't a generic 🙃